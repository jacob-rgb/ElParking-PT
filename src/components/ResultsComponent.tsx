import { replaceSpecialCharacters } from "../helpers/questions";
import { Result, ResultsBox, Subtitle } from "../styled/styledComponents";

interface Props {
    resultsList: any[],
    reverse?:boolean
}

export const ResultsComponent = ({resultsList, reverse = false}: Props) => {
    return (
      <ResultsBox>
          {
              resultsList.sort((a:any,b:any) => {return reverse ?  b.id - a.id : a.id - b.id}).map((answer:any) => (
                  <Result key={answer.question}>
                      <Subtitle>{replaceSpecialCharacters( answer?.question) }</Subtitle>
                      {
                          answer.correct_answer === answer.selectedAnswer 
                          ? (
                              <p className="aligned" style={{color:'green'}}><i style={{color:'green', fontSize:'24px'}} className='bx bxs-check-circle'></i> {replaceSpecialCharacters(answer.selectedAnswer)}</p>
                          )
                          : (
                              <>
                                <p className="aligned"><i style={{color:'red',fontSize:'24px'}} className='bx bxs-x-circle'></i>  <span style={{color:'red'}}> { replaceSpecialCharacters(answer.selectedAnswer)}</span></p>
                                <p className="aligned" style={{color:'gray'}}>Right answer was: { replaceSpecialCharacters(answer.correct_answer) }</p>
                              </>
                          )
                      }
                  </Result>
              ))
          }
      </ResultsBox>
    )
}
