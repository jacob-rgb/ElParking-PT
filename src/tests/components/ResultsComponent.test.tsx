import {ResultsComponent} from '../../components/ResultsComponent';
import { shallow, configure, mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });

const resultsArray = [
    {
        id:0,
        category: "Entertainment: Film",
        selectedAnswer:"Michael Keaton",
        type: "multiple",
        difficulty: "easy",
        question: "Who starred as Bruce Wayne and Batman in Tim Burton&#039;s 1989 movie &quot;Batman&quot;?",
        correct_answer: "Michael Keaton",
        incorrect_answers: [
            "George Clooney",
            "Val Kilmer",
            "Adam West"
        ]
    },
    {
        id:1,
        category: "Geography",
        selectedAnswer:"Hungarian",
        type: "multiple",
        difficulty: "easy",
        question: "All of the following are classified as Finno-Ugric languages EXCEPT:",
        correct_answer: "Samoyedic",
        incorrect_answers: [
            "Hungarian",
            "Finnish",
            "Estonian"
        ]
    },
]

describe("test on <ResultsComponent />", () => {

    let tree;
    
    
    test('Should correctly display the component ',async () => {
        tree = render(< ResultsComponent resultsList={resultsArray} />);   
        expect(tree).toMatchSnapshot();
    });

    test('Icon on first answer is correct, so should be green', () => {
        tree = render(< ResultsComponent resultsList={resultsArray} />);
        const firstPColorStyle = tree.container.querySelector("p")?.style.color;
        expect(firstPColorStyle).toBe('green')
    });

    test('Icon on second answer is incorrect, so should be red', () => {
        tree = render(< ResultsComponent resultsList={resultsArray} />);
        const secondPColorStyle = tree.container.querySelectorAll("i");
        expect(secondPColorStyle[1].style.color).toBe('red');
    });

    test('With the property reverse, The list should be reversed', () => {
        tree = render(< ResultsComponent resultsList={resultsArray} reverse />);
        const fitsrPText = tree.container.querySelector("p span")?.innerHTML;      
        expect(fitsrPText).toMatch(/Hungarian/);
    })
})