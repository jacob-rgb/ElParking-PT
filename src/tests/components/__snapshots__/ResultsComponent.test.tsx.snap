// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <ResultsComponent /> Should correctly display the component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pVTFL jTcbds"
      >
        <div
          class="sc-jrQzAO cQTAty"
        >
          <span
            class="sc-eCImPb iEwKXH"
          >
            Who starred as Bruce Wayne and Batman in Tim Burton's 1989 movie 'Batman'?
          </span>
          <p
            class="aligned"
            style="color: green;"
          >
            <i
              class="bx bxs-check-circle"
              style="color: green; font-size: 24px;"
            />
             
            Michael Keaton
          </p>
        </div>
        <div
          class="sc-jrQzAO cQTAty"
        >
          <span
            class="sc-eCImPb iEwKXH"
          >
            All of the following are classified as Finno-Ugric languages EXCEPT:
          </span>
          <p
            class="aligned incorrect-icon"
          >
            <i
              class="bx bxs-x-circle"
              style="color: red; font-size: 24px;"
            />
              
            <span
              style="color: red;"
            >
               
              Hungarian
            </span>
          </p>
          <p
            class="aligned"
            style="color: gray;"
          >
            Right answer was: 
            Samoyedic
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pVTFL jTcbds"
    >
      <div
        class="sc-jrQzAO cQTAty"
      >
        <span
          class="sc-eCImPb iEwKXH"
        >
          Who starred as Bruce Wayne and Batman in Tim Burton's 1989 movie 'Batman'?
        </span>
        <p
          class="aligned"
          style="color: green;"
        >
          <i
            class="bx bxs-check-circle"
            style="color: green; font-size: 24px;"
          />
           
          Michael Keaton
        </p>
      </div>
      <div
        class="sc-jrQzAO cQTAty"
      >
        <span
          class="sc-eCImPb iEwKXH"
        >
          All of the following are classified as Finno-Ugric languages EXCEPT:
        </span>
        <p
          class="aligned incorrect-icon"
        >
          <i
            class="bx bxs-x-circle"
            style="color: red; font-size: 24px;"
          />
            
          <span
            style="color: red;"
          >
             
            Hungarian
          </span>
        </p>
        <p
          class="aligned"
          style="color: gray;"
        >
          Right answer was: 
          Samoyedic
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
